es77x/src/main/scala/tech/beshu/ror/es/request/context/types/GetEsRequestContext.scala

```diff
@@ -0,0 +1,145 @@
+/*
+ *    This file is part of ReadonlyREST.
+ *
+ *    ReadonlyREST is free software: you can redistribute it and/or modify
+ *    it under the terms of the GNU General Public License as published by
+ *    the Free Software Foundation, either version 3 of the License, or
+ *    (at your option) any later version.
+ *
+ *    ReadonlyREST is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU General Public License for more details.
+ *
+ *    You should have received a copy of the GNU General Public License
+ *    along with ReadonlyREST.  If not, see http://www.gnu.org/licenses/
+ */
+package tech.beshu.ror.es.request.context.types
+
+import cats.data.NonEmptyList
+import cats.implicits._
+import org.elasticsearch.action.ActionResponse
+import org.elasticsearch.action.get.{GetRequest, GetResponse}
+import org.elasticsearch.action.index.IndexRequest
+import org.elasticsearch.action.search.SearchResponse
+import org.elasticsearch.client.node.NodeClient
+import org.elasticsearch.threadpool.ThreadPool
+import tech.beshu.ror.accesscontrol.AccessControlStaticContext
+import tech.beshu.ror.accesscontrol.blocks.BlockContext.GetEsRequestBlockContext
+import tech.beshu.ror.accesscontrol.blocks.metadata.UserMetadata
+import tech.beshu.ror.accesscontrol.domain.{Filter, IndexName}
+import tech.beshu.ror.es.RorClusterService
+import tech.beshu.ror.es.request.AclAwareRequestFilter.EsContext
+import tech.beshu.ror.es.request.DocumentApiOps.DocumentAccessibility.{Accessible, Inaccessible}
+import tech.beshu.ror.es.request.DocumentApiOps.GetApi._
+import tech.beshu.ror.es.request.DocumentApiOps.{GetApi, createSearchRequest}
+import tech.beshu.ror.es.request.RequestSeemsToBeInvalid
+import tech.beshu.ror.es.request.context.ModificationResult.{Modified, ShouldBeInterrupted}
+import tech.beshu.ror.es.request.context.{BaseEsRequestContext, EsRequest, ModificationResult}
+
+import scala.util.{Failure, Success, Try}
+
+class GetEsRequestContext(actionRequest: GetRequest,
+                          esContext: EsContext,
+                          aclContext: AccessControlStaticContext,
+                          clusterService: RorClusterService,
+                          nodeClient: NodeClient,
```

Comment:
let's encapsulate this client and operation on them in ClusterService (it'll be easier to port it to proxy)